InteractiveHtmlBom/generate_interactive_bom.py

```diff
@@ -29,7 +27,13 @@ def to_utf(s):
     from .ecad import get_parser_by_extension
     from .version import version
 
-    app = wx.App()
+    delay_wx = False
+    if os.environ.get('INTERACTIVE_HTML_NO_X11') == 'True':
```

Comment:
X11 is hopefully going to the software heaven soon and will be replaced by modern alternatives (read wayland).
`INTERACTIVE_HTML_BOM_NO_DISPLAY` is more generic.

Also for such flag variables usually just their existence is checked, not specific value. Also helps with case variations and representations of boolean. 

Another thing is the semantics of this variable. It's purpose is to avoid dependency on display server and yet the logic doesn't avoid the dependency, just delays it. It makes no sense to ever set this variable and use `--show-dialog` because that flag implies that you have a display to show that dialog on. I think it's best to just not create wxApp at all if the environment variable is set and exit with error if `--show-dialog` is specified at the same time.

I suggest this fix:
```python
create_wx_app = 'INTERACTIVE_HTML_BOM_NO_DISPLAY' not in os.environ
if create_wx_app:
  import wx
  app = wx.App()
```